### ISSUE TITLE
Issue on login

### ISSUE THREAD

**Issue by apellini** [F5Networks/f5-ansible-f5os#6]
##### COMPONENT NAME

f5networks.f5os.f5os_device_info

### Environment
ansible_user: "{{ ansible_user }}"
ansible_httpapi_password: "{{ ansible_password }}"
ansible_network_os: f5networks.f5os.f5os
ansible_httpapi_use_ssl: true
ansible_httpapi_use_proxy: false
ansible_httpapi_validate_certs: false
ansible_httpapi_port: 8888
ansible_command_timeout: 1800
persistent_log_messages: true

##### ANSIBLE VERSION

```
ansible [core 2.13.6]
  config file = /spimi/NetDevOps/f5_utils/ansible/backup_f5/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.9/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.4 (default, Apr 14 2021, 10:57:48) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
  jinja version = 3.1.2
  libyaml = True
```
##### F5OS VERSION
```
1.3.2-13054
```
##### OS / ENVIRONMENT
N/A

##### SUMMARY
When we try to connect to F5OS device goes in error

##### STEPS TO REPRODUCE

```
- name: Backup F5
  hosts: f5os
  connection: httpapi
  collections:
    - f5networks.f5os
  tasks:
    - name: Collect device information
      f5networks.f5os.f5os_device_info:
        gather_subset:
          - system-info
      environment:
        http_proxy: ""
        https_proxy: ""
```

##### EXPECTED RESULTS
Gathering data from F5

##### ACTUAL RESULTS

```
<IP> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-local-10821gfx97vwk/ansible-tmp-1702399767.5647864-10870-152555949862923/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-local-10821gfx97vwk/ansible-tmp-1702399767.5647864-10870-152555949862923/AnsiballZ_f5os_device_info.py", line 107, in <module>
    _ansiballz_main()
  File "/root/.ansible/tmp/ansible-local-10821gfx97vwk/ansible-tmp-1702399767.5647864-10870-152555949862923/AnsiballZ_f5os_device_info.py", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/root/.ansible/tmp/ansible-local-10821gfx97vwk/ansible-tmp-1702399767.5647864-10870-152555949862923/AnsiballZ_f5os_device_info.py", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.f5networks.f5os.plugins.modules.f5os_device_info', init_globals=dict(_module_fqn='ansible_collections.f5networks.f5os.plugins.modules.f5os_device_info', _modlib_path=modlib_path),
  File "/usr/local/lib/python3.9/runpy.py", line 210, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/local/lib/python3.9/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py", line 1523, in <module>
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py", line 1509, in main
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py", line 1408, in exec_module
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py", line 1442, in execute_managers
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py", line 1249, in exec_module
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py", line 1254, in _exec_module
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py", line 1282, in read_facts
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py", line 1312, in _read_component_facts
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py", line 1324, in read_components_from_device
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/module_utils/client.py", line 26, in wrap
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/module_utils/client.py", line 45, in get
  File "/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible/module_utils/connection.py", line 200, in __rpc__
ansible.module_utils.connection.ConnectionError: {'ietf-restconf:errors': {'error': [{'error-type': 'application', 'error-tag': 'invalid-value', 'error-message': 'uri keypath not found'}]}}
fatal: [device]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-local-10821gfx97vwk/ansible-tmp-1702399767.5647864-10870-152555949862923/AnsiballZ_f5os_device_info.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-local-10821gfx97vwk/ansible-tmp-1702399767.5647864-10870-152555949862923/AnsiballZ_f5os_device_info.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-local-10821gfx97vwk/ansible-tmp-1702399767.5647864-10870-152555949862923/AnsiballZ_f5os_device_info.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.f5networks.f5os.plugins.modules.f5os_device_info', init_globals=dict(_module_fqn='ansible_collections.f5networks.f5os.plugins.modules.f5os_device_info', _modlib_path=modlib_path),\n  File \"/usr/local/lib/python3.9/runpy.py\", line 210, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/local/lib/python3.9/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/local/lib/python3.9/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py\", line 1523, in <module>\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py\", line 1509, in main\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py\", line 1408, in exec_module\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py\", line 1442, in execute_managers\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py\", line 1249, in exec_module\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py\", line 1254, in _exec_module\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py\", line 1282, in read_facts\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py\", line 1312, in _read_component_facts\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/modules/f5os_device_info.py\", line 1324, in read_components_from_device\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/module_utils/client.py\", line 26, in wrap\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible_collections/f5networks/f5os/plugins/module_utils/client.py\", line 45, in get\n  File \"/tmp/ansible_f5networks.f5os.f5os_device_info_payload_rntd_kmr/ansible_f5networks.f5os.f5os_device_info_payload.zip/ansible/module_utils/connection.py\", line 200, in __rpc__\nansible.module_utils.connection.ConnectionError: {'ietf-restconf:errors': {'error': [{'error-type': 'application', 'error-tag': 'invalid-value', 'error-message': 'uri keypath not found'}]}}\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

```

**Comment by pgouband** [comment]
Hi @apellini,

We did a test on Velos and rSeries without any problem.
It's maybe something related to your environment.

**Comment by apellini** [comment]
I not agree because API Auth seems working:

Correct user:
{'ietf-restconf:errors': {'error': [{'error-type': 'application', 'error-tag': 'invalid-value', 'error-message': 'uri keypath not found'}]}}

Wrong user:
 {'ietf-restconf:errors': {'error': [{'error-type': 'protocol', 'error-tag': 'access-denied'}]}}

But seems that rSeries not support API Call.

**Comment by pgouband** [comment]
Hi @apellini,

Unfortunately, we tried to reproduce your issue but we are not able to do it.
So could be something related to the device, the device configuration or your environment.
If you want to go forward, I suggest you open a ticket via https://my.f5.com

**Comment by jewettg** [comment]
I am confirming what @apellini found.   This is MOST DEFINITELY a bug.  

```
TASK [Move tenant f5-r5900-prod-a to configured state] *************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.module_utils.connection.ConnectionError: {'ietf-restconf:errors': {'error': [{'error-type': 'application', 'error-tag': 'invalid-value', 'error-message': 'uri keypath not found'}]}}
```

When I force the collection to v1.4.0, everything works, and if I go above to v1.5.0 or v1.6.0 it blows up.
We are using F5OS 1.5.0-5781.

I can also confirm that the REST API authentication is working and responding with the correct credentials and fails with the wrong credentials. 

**INCORRECT CREDENTIALS:**
```
curl -k -X GET -H "Accept: application/yang-data+json" -u admin:admin https://localhost:8888/restconf//data/openconfig-platform:components/component=platform/state
{  "ietf-restconf:errors": {    "error": [      {        "error-type": "protocol",        "error-tag": "access-denied"      }    ]  }}
```

**CORRECT CREDENTIALS:**
```
curl -k -X GET -H "Accept: application/yang-data+json" -u admin:***** https://localhost:8888/restconf//data/openconfig-platform:components/component=platform/state
{  "openconfig-platform:state": {    "description": "r5900",     "serial-no....   } }
```

**Comment by marlonfrank30** [comment]
> I am confirming what @apellini found. This is MOST DEFINITELY a bug.
> 
> ```
> TASK [Move tenant f5-r5900-prod-a to configured state] *************************
> An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.module_utils.connection.ConnectionError: {'ietf-restconf:errors': {'error': [{'error-type': 'application', 'error-tag': 'invalid-value', 'error-message': 'uri keypath not found'}]}}
> ```
> 
> When I force the collection to v1.4.0, everything works, and if I go above to v1.5.0 or v1.6.0 it blows up. We are using F5OS 1.5.0-5781.
> 
> I can also confirm that the REST API authentication is working and responding with the correct credentials and fails with the wrong credentials.
> 
> **INCORRECT CREDENTIALS:**
> 
> ```
> curl -k -X GET -H "Accept: application/yang-data+json" -u admin:admin https://localhost:8888/restconf//data/openconfig-platform:components/component=platform/state
> {  "ietf-restconf:errors": {    "error": [      {        "error-type": "protocol",        "error-tag": "access-denied"      }    ]  }}
> ```
> 
> **CORRECT CREDENTIALS:**
> 
> ```
> curl -k -X GET -H "Accept: application/yang-data+json" -u admin:***** https://localhost:8888/restconf//data/openconfig-platform:components/component=platform/state
> {  "openconfig-platform:state": {    "description": "r5900",     "serial-no....   } }
> ```

Hi My name is Marlon Frank and I am a solutions engineer at F5. I was troubleshooting this with Greg today. I can confirm that there is a change in the way the API is accessed from 1.4.0 to 1.5.0 on our git repository.....

**Comment by apellini** [comment]
For now we have these F5OS versions:

Product Name: rSeries 2800
Base OS Version: 1.3.2-13054
Service Version: 1.3.2-13054

I think that I need to upgrade on what version I have to go to be compliant?

**Comment by Omripresent** [comment]
To add to this issue I'm seeing a similar problem when using the `f5os_tenant_image` module running F5OS version 1.5.1 on r5900.

Tested both collection versions 1.5.0 and 1.6.0 seeing the exact same result, while 1.4.0 seems to be working correctly.

**Comment by luruiqiang** [comment]
I met the same issue, too. my collection version is 1.6.0, F5OS version is 1.5.1 (r4600)

**Comment by RemcoAA** [comment]
I had the same errors with:

Product Name: rSeries 4600
Base OS Version: 1.5.1-12283
Service Version: 1.5.1-12283
F5OS collection: 1.8.0

but after upgrading to: 
Product Name: rSeries 4600
Base OS Version: 1.7.0-8741
Service Version: 1.7.0-8741
F5OS collection: 1.8.0

I do not have the issue anymore.

Has this bug been solved?

**Comment by RavinderReddyF5** [comment]
Hi @RemcoAA , yes we have added few `f5os_device_info` module changes, thanks for confirming its working

**Comment by RavinderReddyF5** [comment]
Hi @jewettg @Omripresent ,
Issue fixed in v1.10.1 collection, please verify
